src/coreclr/jit/hwintrinsic.cpp

```diff
@@ -285,7 +285,24 @@ NamedIntrinsic HWIntrinsicInfo::lookupId(Compiler*         comp,
 
     bool isIsaSupported = comp->compSupportsHWIntrinsic(isa);
 
-    if ((strcmp(methodName, "get_IsSupported") == 0) || (strcmp(methodName, "get_IsHardwareAccelerated") == 0))
+    bool isHardwareAccelerated = (strcmp(methodName, "get_IsHardwareAccelerated") == 0);
+    if (isHardwareAccelerated)
+    {
+        // Special case: Some of Vector128/256 APIs are hardware accelerated with Sse1 and Avx1,
+        // but we want IsHardwareAccelerated to return true only when all of them are (there are
+        // still can be cases where e.g. Sse41 might give an additional boost for Vector128, but it's
+        // not important enough to bump the minimal Sse version here)
+        if (strcmp(className, "Vector128") == 0)
+        {
+            isa = InstructionSet_SSE2;
+        }
+        else if (strcmp(className, "Vector256") == 0)
+        {
+            isa = InstructionSet_AVX2;
+        }
+    }
```

Comment:
You're going to want `#if defined(TARGET_XARCH)` around this (and likely a comment saying that `TARGET_ARM64` is already correct) as the file is shared

Reply:
Otherwise the change looks correct 👍

Reply:
yeah I was going to ask, in what circumstances does this return true on Arm64?

Reply:
@tannergooding just curious about Q above.

Reply:
@danmoseley `Arm64` is "already correct" as it only needs to consider `AdvSimd` (which is a baseline instruction set)

So, unless the user explicitly disables `AdvSimd` via the environment variable meant for testing; `Vector64/128` will always return `true` and `Vector256` will be `false`

For x86/x64, `Vector64` is always `false` and `Vector128` will be true if `Sse2` is supported (a baseline instruction set, so again only if they explicitly disable the support via the env variable meant for testing); and `Vector256` is true if `Avx2` is supported (nearly 10 years old at this point, should generally be for most customers)

Reply:
Got it - I did'nt realize that `AdvSimd` was baseline for us.